### PR TITLE
Add support for HyperKitty archives, richtext messages, and attachments

### PR DESCRIPTION
**Pull request by LinqLover**
This PR rescues access to the Squeak mailing lists which [have recently been moved to a new archive frontend](https://lists.squeakfoundation.org/archives/list/squeak-dev@lists.squeakfoundation.org/thread/WYGAF674I7EFCNV3EYLKXCOUMWF2HS2W/), HyperKitty: <https://lists.squeakfoundation.org/archives/>.

For this, URLs to affected lists are updated, and the download/processing infrastructure is enabled to handle the web interface and file formats by HyperKitty. As HyperKitty finally provides true MIME messages to us, I also took the chance and added support for accessing attachments and formatted messages text. Not all mailing lists are hosted using HyperKitty, though, so support for pipermail archives is preserved as well. The new capabilities are also provided for pipermail archives.

## TLDR - todos:

- [ ] review sources diff at discretion (see [relevant patches](#relevant-patches))
- [ ] approve reported performance (see [performance](#performance)]
- [ ] backport necessary patches to older versions of Squeak (see [compatibility](#compatibility))

## Performance

Both downloading and processing messages from HyperKitty is slower than from pipermail because the new mbox files include all attachments inline. Downloading full `squeak-dev` from HyperKitty takes about 60 minutes on my machine (40 mbit/s down, 10 mbit/s up, ping 25 ms). Please let me know if further benchmarks are required. We don't have many options to tweak this though. In theory, we could try combining sources from HyperKitty and pipermail to improve the performance, or we could experiment with parallel network connections, but I personally do not think it is worth the effort for this rare use case.

The interactive use case for SqueakInboxTalk, on the other hand, is a few seconds faster than before, as HyperKitty allows us to download files for individual days rather than for entire months.

## Compatibility

The changes are made backward-compatible for Squeak 5.3. Rich text falls back to plain text if `MailMessage>>#bodyTextFormattedRich` (from Network-ct.270/Network-ct.278, Squeak 6.1Alpha) is missing.

However, prior versions of Squeak contain some bugs and lack some robustness for parsing mail messages, leading to a debugger when processing all lists. To solve this issue, my recommendation would be to port back the patches for `MIMEDocument>>#asSendableBodyText`, `MIMEDocument>>#initializeFromString:`, and `MIMEDocument class>>#parseParts:withSeparator:` back to Squeak 6.0 and Squeak 5.3. Alternatively, we could provide them through a compatibility package [like in Squot](https://github.com/hpi-swa/Squot/tree/develop/src/SquotCompatibility-Squeak50.package), but I personally would dislike the added complexity.

Given that these patches are installed, all basic use cases (installation, downloading all archives, using the Demo category) do work in a Squeak 5.3 image.

## Relevant Patches

- Step 1: Implement download of lists from hyperkitty archives (cfd22b616d07edf7ec2d5d761e3aa5d5a0c00a07)

   New protocol for piecewise downloads in `SqhPipermailWrapper class>>#downloadListNamed:forTimespans:`. Removed old `#downloadListNamed:forMonthsThat:` (which was introduced in https://github.com/hpi-swa/squeak-history/pull/8).

   Refactor list specs and (piecewise) downloading interface.  
   Add new cuis-dev mailing list.  
   Improve handling of network errors.
 
- Step 2: Update processing of messages (2f4cba7f01efb4af3ab847d2a7bc04462bacccc6) 

   Honor date filters per message. Add `#onlyTimespans:` convenient accessor.
   Update author normalization for hyperkitty messages.
   Improve progress display.
   Make parsing of separators in mbox files more robust.
 
- Step 3: Add `SqhMailWrapper` subclasses for different archive sources (c43205118a76583a1778b742b4b1d7abdfddb735)

   Push down pipermail-specific decoding hacks to `SqhPipermailWrapper`.
 
- Step 4: Add support for rich text and attachments (25776fb75e5643b5090fec1e2314ce613ea47860) 

   Add `#richText`, `#plainText`, and `#rawText` on `SqhMailWrapper`. Deprecate `#bodyText` and update senders.
   Add `#attachmentNames`, `#attachments`, and `#attachmentAt:` on `SqhMailWrapper`.

- Addenda (all newer commits)

This PR also iterates on and thus merges #8, #9, #10, and #13.

image

### CODE REVIEW

**Comment by marceltaeumel** [comment]
I closed PRs 9, 10, and 13. I merged 8 by accident. Please update this PR so I can merge it.
